interfaces/tests/lock_ut/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+cmake_minimum_required(VERSION 2.8.11)
+
+set(theseTestsName lock_ut)
+
+set(${theseTestsName}_test_files
+    ${theseTestsName}.c
+)
+
+set(${theseTestsName}_c_files
+    ${LOCK_C_FILE}
+)
+
+set(${theseTestsName}_h_files
+    ../../inc/lock.h
+)
+
+build_c_tests(${theseTestsName} ON "tests/azure_c_pal/int/" azure_c_pal)
```

Comment:
Shouldn't link to azure_c_pal since this is a unittest (probably just pal_interfaces)